### PR TITLE
feat: export useStore hook using the isolated context

### LINKED ISSUE #26: expose context from create isolation
Because useStore by default uses the default context, when working in isolation, it's not possible to utilize hooks like useHydrateAtoms, as they all rely on useStore. Consequently, they would hydrate a different store from a different context.

I think it would be correct and beneficial either to directly export the context or to offer a custom useStore function, allowing other hooks to receive the store as an option.

I'm willing to submit a pull request if necessary.

### PR DESCRIPTION
**Pull request by marcocondrache**
fixes #26

### CODE REVIEW

File: src/createIsolation.tsx [hunk 1]
```diff
@@ -36,29 +36,26 @@ export function createIsolation() {
     );
   };
 
-  const useAtom = ((anAtom: any, options?: any) => {
+  const useStore = () => {
```
Comment (dai-shi) [inline src/createIsolation.tsx]: Do we not want to keep the original signature of useStore?

Comment (marcocondrache) [reply src/createIsolation.tsx]: fixed

File: src/createIsolation.tsx [hunk 2]
```diff
@@ -36,29 +37,26 @@ export function createIsolation() {
...
   };
 
-  const useAtom = ((anAtom: any, options?: any) => {
+  const useStore = ((options?: any) => {
     const store = useContext(StoreContext);
-    if (!store) {
-      throw new Error('Missing Provider from createIsolation');
-    }
+    if (!store) throw new Error('Missing Provider from createIsolation');
+    return options?.store || store;
```
Comment (dai-shi) [inline src/createIsolation.tsx]: Thanks.
How about this for consistency?
```suggestion
    return useStoreOrig({ store, ...options });
```

Comment (marcocondrache) [reply src/createIsolation.tsx]: Thank you. I can't believe I missed that 😄

**Review by dai-shi — APPROVED** [review]
LGTM!

@yf-yang When you have a chance, would you be able to merge this and publish a new version?

**Pull request merged**